### PR TITLE
[Feature/51601] Both platforms: Updated layout and flow of countries consent

### PR DESCRIPTION
**Pull request by jonasofrich**
IMG_0212
IMG_0211

### CODE REVIEW

File: NDB.Covid19/NDB.Covid19.Droid/Resources/layout/countries_consent.xml [hunk 1]
```diff
@@ -98,15 +98,31 @@
...
                     android:layout_marginVertical="@dimen/defaultVerticalMargin"
+                    android:layout_marginHorizontal="@dimen/defaultHorizontalMargin"
                     android:layout_width="match_parent"
-                    android:layout_height="wrap_content"
-                    android:layout_centerHorizontal="true"
-                    android:layout_below="@id/explanation_text_textview"
-                    android:id="@+id/consent_button"
+                    android:layout_height="fill_parent"
+                    android:layout_weight="1"
+                    android:id="@+id/consentWithEU_button"
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Resources/layout/countries_consent.xml:114]: Change id to "consentButton_EU"

File: NDB.Covid19/NDB.Covid19.Droid/Resources/layout/countries_consent.xml [hunk 2]
```diff
@@ -98,15 +98,31 @@
...
+                    style="@style/IDPortenButton"/>
+                <Button
+                    android:text="Consent Button"
+                    android:layout_marginVertical="@dimen/defaultVerticalMargin"
+                    android:layout_marginHorizontal="@dimen/defaultHorizontalMargin"
+                    android:layout_width="match_parent"
+                    android:layout_weight="1"
+                    android:layout_height="fill_parent"
+                    android:id="@+id/consentWithoutEU_button"
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Resources/layout/countries_consent.xml:123]: Change id to "consentButton_onlyNorway"

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs [hunk 3]
```diff
@@ -26,7 +26,8 @@ public class CountriesConsentActivity: AppCompatActivity
         private TextView _notificationText;
         private TextView _beAwareText;
         private TextView _consentExplanationText;
-        private Button _consentButton;
+        private Button _consentWithEUButton;
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs:29]: Change to "consentButton_EU"

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs [hunk 4]
```diff
@@ -26,7 +26,8 @@ public class CountriesConsentActivity: AppCompatActivity
...
+        private Button _consentWithoutEUButton;
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs:30]: Change id to "consentButton_onlyNorway"

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs [hunk 5]
```diff
@@ -40,7 +41,8 @@ private void InitLayout()
         {
             //Buttons
             _closeButton = FindViewById<ViewGroup>(Resource.Id.close_cross_btn);
-            _consentButton = FindViewById<Button>(Resource.Id.consent_button);
+            _consentWithEUButton = FindViewById<Button>(Resource.Id.consentWithEU_button);
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs:44]: Change id to "consentButton_EU"

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs [hunk 6]
```diff
@@ -40,7 +41,8 @@ private void InitLayout()
...
+            _consentWithoutEUButton = FindViewById<Button>(Resource.Id.consentWithoutEU_button);
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs:45]: Change id to "consentButton_onlyNorway"

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs [hunk 7]
```diff
@@ -53,7 +55,8 @@ private void InitLayout()
             _consentExplanationText = FindViewById<TextView>(Resource.Id.explanation_text_textview);
 
             //Text initialization
-            _consentButton.Text = BUTTON_TEXT;
+            _consentWithEUButton.Text = EU_CONSENT_NEXT_WITH_CONSENT_BUTTON_TEXT;
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs:58]: EU

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs [hunk 8]
```diff
@@ -53,7 +55,8 @@ private void InitLayout()
...
+            _consentWithoutEUButton.Text = EU_CONSENT_NEXT_WITHOUT_CONSENT_BUTTON_TEXT;
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs:59]: onlyNorway

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs [hunk 9]
```diff
@@ -67,19 +70,14 @@ private void InitLayout()
             _closeButton.ContentDescription = CLOSE_BUTTON_ACCESSIBILITY_LABEL;
 
             //Button click events
-            _closeButton.Click += new StressUtils.SingleClick((sender, e) => ShowAreYouSureToExitDialog(), 500).Run;
-            _consentButton.Click += new StressUtils.SingleClick(async (o, args) =>
+            _closeButton.Click += new StressUtils.SingleClick((sender, e) => ShowAbortDialog(), 500).Run;
+            _consentWithEUButton.Click += new StressUtils.SingleClick((o, args) =>
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs:74]: EU

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs [hunk 10]
```diff
@@ -67,19 +70,14 @@ private void InitLayout()
...
-                {
-                    InvokeNextButtonClick(GoToCountriesQuestionnairePage, OnFail, consentGiven);
-                }
-                else
-                {
-                    InvokeNextButtonClick(GoToLoadingPage, OnFail, consentGiven);
-                }
+                InvokeNextButtonClick(GoToCountriesQuestionnairePage, OnFail, true);
+            }, 500).Run;
+            _consentWithoutEUButton.Click += new StressUtils.SingleClick((o, args) =>
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs:78]: onlyNorway

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs [hunk 11]
```diff
@@ -102,20 +100,14 @@ private void GoToCountriesQuestionnairePage()
...
                 this,
-                ErrorViewModel.REGISTER_LEAVE_HEADER,
-                ErrorViewModel.REGISTER_LEAVE_DESCRIPTION,
-                ErrorViewModel.REGISTER_LEAVE_CONFIRM,
-                ErrorViewModel.REGISTER_LEAVE_CANCEL);
-            if (isOkPressed)
-            {
-                GoToInfectionStatusPage();
-            }
+                EU_CONSENT_ABORT_DIALOG_HEADER,
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs:107]: CONSENT3_ABORT_DIALOG.....
^for all these 3 names

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsent.storyboard [hunk 12]
```diff
@@ -105,7 +100,21 @@
                                                             <color key="titleColor" white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
                                                         </state>
                                                         <connections>
-                                                            <action selector="OnNext:" destination="326" id="6030" eventType="touchUpInside"/>
+                                                            <action selector="OnNextWithEUConsent:" destination="326" eventType="touchUpInside" id="83g-CM-IeP"/>
+                                                        </connections>
+                                                    </button>
+                                                    <button opaque="NO" contentMode="scaleToFill" contentHorizontalAlignment="center" contentVerticalAlignment="center" lineBreakMode="wordWrap" translatesAutoresizingMaskIntoConstraints="NO" id="qUd-pk-VPy" userLabel="NextWithoutEUConsent" customClass="DefaultBorderButton">
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsent.storyboard:106]: Label = NextOnlyNorwayConsent

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsent.storyboard [hunk 13]
```diff
@@ -105,7 +100,21 @@
...
+                                                        <fontDescription key="fontDescription" type="system" pointSize="16"/>
+                                                        <inset key="contentEdgeInsets" minX="10" minY="0.0" maxX="10" maxY="0.0"/>
+                                                        <inset key="titleEdgeInsets" minX="3" minY="0.0" maxX="0.0" maxY="0.0"/>
+                                                        <inset key="imageEdgeInsets" minX="0.0" minY="0.0" maxX="3" maxY="0.0"/>
+                                                        <state key="normal" title="Continue">
+                                                            <color key="titleColor" white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
+                                                        </state>
+                                                        <connections>
+                                                            <action selector="OnNextWithoutEUConsent:" destination="326" eventType="touchUpInside" id="Of7-SE-qbU"/>
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsent.storyboard:117]: Selector = OnNextOnlyNorwayConsent

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsent.storyboard [hunk 14]
```diff
@@ -144,9 +154,10 @@
                         <outlet property="HeaderLabel" destination="mRS-C6-GLl" id="name-outlet-mRS-C6-GLl"/>
                         <outlet property="LookUp_Header" destination="7hc-mI-fcI" id="name-outlet-7hc-mI-fcI"/>
                         <outlet property="LookUp_Text" destination="rNX-yt-9M4" id="name-outlet-rNX-yt-9M4"/>
+                        <outlet property="NextButtonWithEUConsent" destination="Q5l-do-83F" id="P2A-Y6-ypH"/>
+                        <outlet property="NextButtonWithoutEUConsent" destination="qUd-pk-VPy" id="oJK-rI-zUo"/>
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsent.storyboard:158]: NextButtonOnlyNorwayConsent

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs [hunk 15]
```diff
@@ -36,7 +36,8 @@ public override void ViewDidLoad()
             Consent_BeAware_Text.SetAttributedText(COUNTRIES_CONSENT_BE_AWARE_TEXT_CONSENT_BE_AWARE_TEXT);
             Consent_Explanation_Text.SetAttributedText(COUNTRIES_CONSENT_BE_AWARE_TEXT_CONSENT_EXPLANATION_TEXT, StyleUtil.FontType.FontItalic);
 
-            StyleUtil.InitButtonStyling(NextButton, BUTTON_TEXT);
+            StyleUtil.InitButtonStyling(NextButtonWithEUConsent, EU_CONSENT_NEXT_WITH_CONSENT_BUTTON_TEXT);
+            StyleUtil.InitButtonStyling(NextButtonWithoutEUConsent, EU_CONSENT_NEXT_WITHOUT_CONSENT_BUTTON_TEXT);
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs:40]: NextButtonOnlyNorwayConsent

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs [hunk 16]
```diff
@@ -81,14 +77,14 @@ void OnFail()
                 "AuthenticationState.PersonalData was garbage collected (iOS)");
         }
 
-        partial void OnNext(DefaultBorderButton sender)
+        partial void OnNextWithEUConsent(DefaultBorderButton sender)
+        {
+            InvokeNextButtonClick(GoToQuestionnaireCountriesPage, OnFail, true);
+        }
+
+        partial void OnNextWithoutEUConsent(DefaultBorderButton sender)
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs:85]: OnNextOnlyNorwayConsent

File: NDB.Covid19/NDB.Covid19/ViewModels/CountriesConsentViewModel.cs [hunk 17]
```diff
@@ -38,7 +44,13 @@ public class CountriesConsentViewModel
         public static string COUNTRIES_CONSENT_DIALOG_MESSAGE => "COUNTRIES_CONSENT_DIALOG_MESSAGE".Translate();
         public static string COUNTRIES_CONSENT_DIALOG_OK => "COUNTRIES_CONSENT_DIALOG_OK".Translate();
         public static string COUNTRIES_CONSENT_DIALOG_CANCEL => "COUNTRIES_CONSENT_DIALOG_CANCEL".Translate();
-        
+        public static string EU_CONSENT_NEXT_WITH_CONSENT_BUTTON_TEXT => "EU_CONSENT_NEXT_WITH_CONSENT_BUTTON_TEXT".Translate();
+        public static string EU_CONSENT_NEXT_WITHOUT_CONSENT_BUTTON_TEXT => "EU_CONSENT_NEXT_WITHOUT_CONSENT_BUTTON_TEXT".Translate();
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19/ViewModels/CountriesConsentViewModel.cs:48]: EU_CONSENT_NEXT_ONLYNORWAY_CONSENT_BUTTON_TEXT

**Review by Frederik-S-N-dk — CHANGES_REQUESTED** [review]

File: NDB.Covid19/NDB.Covid19.Droid/Resources/layout/countries_consent.xml (hunk 1 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19.Droid/Resources/layout/countries_consent.xml]: done

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs (hunk 3 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs]: Went with _consentButtonEU to follow convention

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs (hunk 4 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs]: Went with _consentButtonOnlyNorway to follow convention

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs (hunk 5 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/CountriesConsentActivity.cs]: same as above

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsent.storyboard (hunk 13 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsent.storyboard]: Done

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsent.storyboard (hunk 14 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsent.storyboard]: Done

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs (hunk 15 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs]: Done

File: NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs (hunk 16 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19.iOS/Views/AuthenticationFlow/CountriesConsentViewController.cs]: Done

File: NDB.Covid19/NDB.Covid19/ViewModels/CountriesConsentViewModel.cs (hunk 17 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19/ViewModels/CountriesConsentViewModel.cs]: Used EU_CONSENT_NEXT_ONLY_NORWAY_CONSENT_BUTTON_TEXT for consistency

File: NDB.Covid19/NDB.Covid19/ViewModels/CountriesConsentViewModel.cs [hunk 18]
```diff
@@ -38,7 +44,13 @@ public class CountriesConsentViewModel
         public static string COUNTRIES_CONSENT_DIALOG_MESSAGE => "COUNTRIES_CONSENT_DIALOG_MESSAGE".Translate();
         public static string COUNTRIES_CONSENT_DIALOG_OK => "COUNTRIES_CONSENT_DIALOG_OK".Translate();
         public static string COUNTRIES_CONSENT_DIALOG_CANCEL => "COUNTRIES_CONSENT_DIALOG_CANCEL".Translate();
-        
+        public static string EU_CONSENT_NEXT_WITH_CONSENT_BUTTON_TEXT => "EU_CONSENT_NEXT_WITH_CONSENT_BUTTON_TEXT".Translate();
```
Comment (Frederik-S-N-dk) [inline NDB.Covid19/NDB.Covid19/ViewModels/CountriesConsentViewModel.cs:47]: same comment as previously

**Comment by pavshr** [comment]
The dialog on the first screen, when does it appear?

File: NDB.Covid19/NDB.Covid19/ViewModels/CountriesConsentViewModel.cs (hunk 18 above)
Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19/ViewModels/CountriesConsentViewModel.cs]: 🤦 Fixed

**Comment by jonasofrich** [comment]
> The dialog on the first screen, when does it appear?

@pavshr when clicking the X in the upper right corner

**Review by Frederik-S-N-dk — APPROVED** [review]

**Pull request merged**